src/Controller/JobsController.php

```diff
@@ -0,0 +1,296 @@
+                        'results' => [
+                            'error' => true,
+                            'message' => $e->getMessage()
+                        ]
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * updateJobs
+     *
+     * @param  mixed $request
+     * @param  mixed $id
+     * @return Response
+     */
+    public function updateJob(Request $request, int $id): Response
+    {
+        try {
+            $requestParams = $request->query->all();
+
+            $updateResult = $this->jobsRepository->update($id, $requestParams);
+
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'updated' => $updateResult,
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * getJobList
+     *
+     * @return Response
+     */
+    public function getJobList(): Response
+    {
+        $jobs = $this->jobsRepository->findAll();
+
+        $name = [];
+
+        foreach ($jobs as $job) {
+            $name[] = $job->getName();
+        }
+
+        return new JsonResponse(
+            [
+                'results' => [
+                    'jobs' => [
+                        'name' => $name,
+                    ],
+                    'error' => false,
+                ]
+            ]
+        );
+    }
+
+    /**
+     * deleteJob
+     *
+     * @param  mixed $id
+     * @return Response
+     */
+    public function deleteJob(int $id): Response
+    {
+        $job = $this->jobsRepository->find($id);
+
+        if (is_null($job)) {
+            throw new \InvalidArgumentException('Could not find job with id: ' . $id);
+        }
+
+        try {
+            $this->jobsRepository->remove($job);
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'deleted' => $id,
+                        'error' => false,
+                    ]
+                ]
+            );
+        } catch (Exception $e) {
+            return new JsonResponse(
+                [
+                    'results' => [
+                        'error' => true,
+                        'message' => $e->getMessage()
+                    ]
+                ]
+            );
+        }
+    }
+
+    /**
+     * getjobName
+     *
+     * @param  mixed $name
+     * @return Response
+     */
+    public function getjobName(string $name): Response
+    {
+        try {
+            $result = $this->jobsRepository->getJobName($name);
+
+            return new JsonResponse(
+                [
+                    'results' => [
```

Comment:
incomplete information about job. Missing the rest of the fields and company information. The response must be like at function getJobById()